dll/win32/shlwapi/ordinal.c

```diff
@@ -5670,6 +5699,70 @@ HRESULT WINAPI SHPropertyBag_ReadRECTL(IPropertyBag *ppb, LPCWSTR pszPropName, R
     return SHPropertyBag_ReadLONG(ppb, szBuff, &prcl->bottom);
 }
 
+/**************************************************************************
+ *  SHPropertyBag_ReadGUID (SHLWAPI.505)
+ */
+HRESULT WINAPI SHPropertyBag_ReadGUID(IPropertyBag *ppb, LPCWSTR pszPropName, GUID *pguid)
+{
+    HRESULT hr;
+    BOOL bRet;
+    VARIANT vari;
+
+    TRACE("%p %s %p\n", ppb, debugstr_w(pszPropName), pguid);
+
+    if (!ppb || !pszPropName || !pguid)
+    {
+        ERR("%p %s %p\n", ppb, debugstr_w(pszPropName), pguid);
+        return E_INVALIDARG;
+    }
+
+    hr = SHPropertyBag_ReadType(ppb, pszPropName, &vari, VT_EMPTY);
+    if (FAILED(hr))
+    {
+        ERR("%p %s %p\n", ppb, debugstr_w(pszPropName), pguid);
+        return hr;
+    }
+
+    if (V_VT(&vari) == (VT_UI1 | VT_ARRAY)) /* Byte Array */
+        bRet = VariantToBuffer(&vari, pguid, sizeof(*pguid));
+    else if (V_VT(&vari) == VT_BSTR)
+        bRet = GUIDFromStringW(V_BSTR(&vari), pguid);
+    else
+        bRet = TRUE; /* This is by design */
```

Comment:
Maybe, but what could be the reason behind this?
Is this behaviour tested by your tests?

Reply:
Reason I don't know. Yeah, tested.

Reply:
OK, but in that case, what does the data buffer pointed by `pguid` contains on return? Garbage? or all zeroes?

Reply:
It's a bug in WinXP/Win2k3. Win10 won't reproduce this behavior.
It's correctly returning a GUID value.

Reply:
Hmmmm.... I don't know, but I would prefer in that case, that we actually fix the bug and return whatever correct GUID in that case (a null one perhaps).
Cc @ThFabba @learn-more for some advice on that.